src/tests/utils/test_utils.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+import re
+import numpy as np
+from numpy.random import default_rng
+from scipy.special import expit
+
+from med_bench.get_simulated_data import simulate_data
+from med_bench.utils.utils import _check_input
+
+
+rg = default_rng(5)
+n = 5
+dim_x = 3
+
+x = rg.standard_normal(n * dim_x).reshape((n, dim_x))
+binary_m_or_t = rg.binomial(1, 0.5, n).reshape(-1, 1)
+y = rg.standard_normal(n).reshape(-1, 1)
+
+
+testdata = [
+    (x, binary_m_or_t, binary_m_or_t, x, "Multidimensional y (outcome)"),
+    (y, x, binary_m_or_t, x, "Multidimensional t (exposure)"),
+    (y, x[0], binary_m_or_t, x, "Only a binary t (exposure)"),
+    (y, np.vstack([binary_m_or_t, binary_m_or_t]), binary_m_or_t, x, 
+        "same number of observations"),
+    (y, binary_m_or_t, np.vstack([binary_m_or_t, binary_m_or_t]), x, 
+        "same number of observations"),
+    (y, binary_m_or_t, binary_m_or_t, np.vstack([x, x]),
+        "same number of observations"),
+    (y, binary_m_or_t, x, x, "Multidimensional m (mediator)"),
+    (y, binary_m_or_t, x[:, 0], x, "a binary one-dimensional m"),
```

Comment:
m is not binary in that case, is it ?

Reply:
in fact, you can specify whether ou want a binary 1D-M or not with the last argument of the function. This test checks that when you specify 'binary' it raises if you provide a non-binary M (the first dimension of x for the test). Should I add comments somewhere in the code to clarify that?

Reply:
Sorry, I misunderstood the whole thing.
A samll comment about what the function is cheking will help me next time ;-)

Reply:
added in the docstring. Ok?

Reply:
Sorry, where ? My suggestion was to document the test too.